src/lib/components/input/input.js

```diff
@@ -1,83 +1,77 @@
-import { handleInputChange } from './utils.ignore';
-import styles from './input.module.scss';
+import React from "react";
+import PropTypes from "prop-types";
+import {
+  componentPropTypes,
+  defaultComponentPropTypes,
+} from "../../utils/componentPropTypes";
+import { handleInputChange } from "../oldInput/utils.ignore";
+import styles from "./input.module.scss";
 
-class Input extends PureComponent {
-  static propTypes = {
-    ...componentPropTypes,
-    placeholder: PropTypes.string,
-    disabled: PropTypes.bool,
-    required: PropTypes.bool,
-    name: PropTypes.string.isRequired,
-    value: PropTypes.string,
-    type: PropTypes.string,
-    autoFocus: PropTypes.bool,
-    onChangeFunction: PropTypes.func,
-    // props from withValidation HOC
-    renderValidationError: PropTypes.func,
-    handleValidations: PropTypes.func.isRequired,
-    isValidationOk: PropTypes.func.isRequired,
-  };
+const Input = ({
+  theme,
+  name,
+  value,
+  error,
+  label,
+  className,
+  type,
+  onChangeFunction,
+  handleValidations,
+  helperText,
+  containerClassName,
+  ...inputProps
+}) => {
+  const handleChange = event =>
+    handleInputChange(event, name, handleValidations, onChangeFunction);
 
-  static defaultProps = {
-    ...defaultComponentPropTypes,
-    placeholder: '',
-    disabled: false,
-    required: true,
-    value: '',
-    type: 'text',
-    onChangeFunction: () => {},
-  };
-
-  handleChange = name => event =>
-    handleInputChange(event, name, this.props.handleValidations, this.props.onChangeFunction);
-
-  render() {
-    const {
-      containerClassName,
-      className,
-      theme,
-      placeholder,
-      disabled,
-      required,
-      name,
-      value,
-      type,
-      isValidationOk,
-      renderValidationError,
-      autoFocus,
-      // removing props that are not valid for DOM elements
-      // to hand the rest to the input
-      handleValidations,
-      onChangeFunction,
-      getValidationMessage,
-      validationMessage,
-      hasErrorAfterSubmit,
-      // props handed to the input DOM element
-      ...restProps
-    } = this.props;
-
-    return (
-      <div className={`${containerClassName} ${styles[`theme-${theme}`]} ${styles.inputWrapper}`}>
+  return (
+    <div className={styles[`theme-${theme}`]}>
+      <div className={styles.wrapper}>
         <input
-          placeholder={placeholder}
-          className={`${isValidationOk() && styles.error} ${styles.input} ${className}`}
-          value={value}
+          className={`${className} ${error && styles.error}`}
+          type={type || "text"}
+          id={name}
           name={name}
-          type={type}
-          required={required}
-          onChange={this.handleChange(name)}
-          disabled={disabled ? 'disabled' : ''}
-          autoFocus={autoFocus}
-          {...restProps}
+          onChange={handleChange}
+          value={value}
+          {...inputProps}
```

Comment:
I don't know if they need to be or not but the props `...spread` items do they need to be in `proptypes`?

Reply:
1. The whole thing with the proptypes is … optional, the way we use it, so it's always a nice to have, and nothing breaks if we don't do it 100%. 
2. I think the idea is that we just pass all the rest of the props, whatever they might be. If that is the case, we don't know what's in there. 

So … I'd leave it like that, unless we decide we want to be more strict about things like that.